flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/loading/DefaultLoadingWeight.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.scheduler.loading;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.util.Preconditions;
+
+import javax.annotation.Nonnull;
+
+import java.util.Objects;
+
+/** The default implementation of {@link LoadingWeight} based on task count loading. */
+@Internal
+public class DefaultLoadingWeight implements LoadingWeight {
+
+    private float tasks;
+
+    DefaultLoadingWeight(float tasks) {
+        Preconditions.checkArgument(tasks >= 0.0f);
+        this.tasks = tasks;
+    }
+
+    public void incLoading() {
+        this.tasks += 1.0f;
+    }
+
+    @Override
+    public float getLoading() {
+        return tasks;
+    }
+
+    @Override
+    public LoadingWeight merge(LoadingWeight other) {
+        if (other == null) {
+            return LoadingWeight.ofDefaultLoadingWeight(this.tasks);
```

Comment:
The merge method should new DefaultLoadingWeight directly.

If  `LoadingWeight.ofDefaultLoadingWeight` is refactored in the future, it create other type of `LoadingWeight`. It will create other LoadingWeight when `DefaultLoadingWeight` is merged.

Also, if other == null, could we return this directly?